### PR TITLE
DIFM: Adds social profiles step

### PR DESCRIPTION
**Pull request by aneeshd16**
#### Changes proposed in this Pull Request

* This PR builds on top of #61835.
* This is the UI for "Social Profiles" step for the new DIFM flow. The PR does not implement the complete flow. The new step is accessible when the `signup/redesigned-difm-flow` feature flag is enabled. However, the end-to-end flow will not be testable.
* The "Continue" CTA dispatches an action that saves data to the redux store and submits the step.
* The "Skip" CTA removes data from the redux store and then submits the step.
image

#### Testing instructions

* Go to `/start/do-it-for-me/?flags=signup/redesigned-difm-flow`
* Confirm that the UI matches the screenshot shared above.

Related to 0-as-1201933307121196/f

### CODE REVIEW

File: client/signup/steps/social-profiles/social-profiles.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+						<FormInput
+							name={ key }
+							id={ key }
+							value={ formValues[ key as SocialProfile ] }
+							placeholder={ socialProfileOption.placeholder }
+							onChange={ onChange }
+						/>
+					</TextInputWrapper>
+				</FormFieldset>
+			) ) }
```
Comment (ddc22) [inline client/signup/steps/social-profiles/social-profiles.tsx]: Why do we need this abstraction?

We can probably encapsulate this code into its own component and just do.

```jsx
<SocialProfileField  name="facebook" onChange={onChange} value={formValues['facebook']} / >
<SocialProfileField  name="instagram" onChange={onChange} value={} / >
<SocialProfileField  name="apple" onChange={onChange} value={} / >
<SocialProfileField  name="google" onChange={onChange} value={} / >
```

I feel like our codebase is growing with this pattern where we build our data structure in an array before rendering and then loop over it which is a very jquery like pattern. 

But by doing this we lose the expressiveness of JSX where we can get a look and feel extremely close to declarative HTML while completely letting go the complexity required to maintain a for loop.

This is again a personal opinion but what do you think?

File: client/lib/signup/step-actions/index.js [hunk 2]
```diff
@@ -349,17 +350,20 @@ function getDIFMLiteCartItemFromDependencies( dependencies ) {
 		displayPhone,
 		displayAddress,
 	} = dependencies;
+
+	const socialProfiles = getSocialProfiles( reduxStore.getState() );
```
Comment (ddc22) [inline client/lib/signup/step-actions/index.js]: Maybe we can have a difm lite selector which transforms our state tree to a flat data structure of fields to be submitted. This way the step-actions module does not need to know about the UI separated sections of the state tree,

**Review by ddc22 — COMMENTED** [review]
Change looks nice! Have a small UI glitch below!
Can we possibly make the space around the icon more like a square? :)
image

Also left some comments about the code below.

File: client/lib/signup/step-actions/index.js (hunk 2 above)
Comment (aneeshd16) [reply client/lib/signup/step-actions/index.js]: Good idea. For now, I'll remove this change from the PR, and we can continue this part in a follow-up PR.

File: client/signup/steps/social-profiles/social-profiles.tsx (hunk 1 above)
Comment (aneeshd16) [reply client/signup/steps/social-profiles/social-profiles.tsx]: Thanks for the feedback! I have made the changes, can you please have a look?

**Review by ddc22 — APPROVED** [review]
Nice refactor!
I also cannot reproduce that border that appeared.

However, in some brightnesses on my monitor I can see a faint tint of that icon padding being disproportionate.
I have darkened what I see so that it is clearly visible.
image

However this seems like an extremely minor glitch we can iterate on later!

So LGTM 🚢

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7199515

Thank you @aneeshd16 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7199515).

**Pull request merged**